src/net/java/textilej/parser/markup/textile/phrase/SimpleTextilePhraseModifier.java

```diff
@@ -49,7 +49,15 @@ public SimpleTextilePhraseModifier(String delimiter, SpanType spanType, boolean
 	@Override
 	protected String getPattern(int groupOffset) {
 		String quotedDelimiter = quoteLite(getDelimiter());
-		
+
+		// special casing to avoid breaking freenet keys:
+		if (this.delimiter == "-") {
```

Comment:
Strings have to be compared using `.equals()`, so `if (delimiter.equals("-"))` would be correct—unless `delimiter` can be `null`.